### DISCUSSION TITLE
Little Help #2

### DISCUSSION THREAD

**Discussion by copycatcodex** [masterking32/MasterHamsterKombatBot#312] [Q&A] [answered]
I ran the bot this morning, and when I entered the playground section of the game an error like this occurred, the previous day it was normal, is there something wrong?

[Master HamsterKombat Bot][ERROR] 2024-09-01 11:29:21 [Random 1] **Getting Gangs Wars key...**
[Master HamsterKombat Bot][ERROR] 2024-09-01 11:29:21 [Random 1] **Error: HTTPSConnectionPool(host='api.gamepromo.io', port=443): Max retries exceeded with url: /promo/login-client (Caused by ProxyError('Unable to connect to proxy', OSError('Tunnel connection failed: 403 Forbidden')))**
[Master HamsterKombat Bot][ERROR] 2024-09-01 11:29:21 [Random 1] **Error: HTTPSConnectionPool(host='api.gamepromo.io', port=443): Max retries exceeded with url: /promo/login-client (Caused by ProxyError('Unable to connect to proxy', OSError('Tunnel connection failed: 403 Forbidden')))**
[Master HamsterKombat Bot][ERROR] 2024-09-01 11:29:21 [Random 1] **Unable to get Gangs Wars key.**

So I temporarily disabled the auto playground setting, afraid there would be a more serious problem

**Comment by Fy0urM** [comment, marked as answer]
> Caused by ProxyError('Unable to connect to proxy'

not a bot error, check your proxy or ethernet connection or link from your pc to proxy server

**Comment by copycatcodex** [comment]
it looks like my proxy is dead, is it safe to run 2 accounts each one bot without a proxy? so I run 2 bots at once

**Comment by Fy0urM** [comment]
in theory, there is nothing wrong if you have 2 accounts on 1 ip, for example, when you are in 1 room with someone and connected to 1 wifi network your devices have the same ip address. but on the bot side it can cause problems because there is a limit on the number of requests to api from 1 ip per second and if you have 2 accounts in 2 bots running in parallel and they simultaneously send a request to api in the best case 1. there is a limit on the number of requests to api with 1 ip per second and if you have 2 accounts in 2 bots running in parallel and they simultaneously send a request to api in the best case 1 bot will get a response from the server and the other will not because it will get a limitation for exceeding the limit of requests. if you have 1 bot and 2 accounts this should not happen because the accounts are executed sequentially one after the other and not in parallel.

**Comment by copycatcodex** [comment]
The explanation is understandable, maybe it's safe to have 1 bot and 2 accounts (even though each account takes a long time because of the long process), and with the same proxy, it's still safe to use.

**Comment by copycatcodex** [comment]
![image](https://github.com/user-attachments/assets/63076ea7-ceca-47ad-8593-5ca930ac9a80)
yes there is something else I want to ask, previously I used a proxy with settings like this, (see picture) and it ran smoothly, until now it died. but when I added https to the second line, or I went back and forth to http. often there was an error in the proxy until finally I didn't give http/s (as in the picture) and it ran normally

Error: HTTPSConnectionPool(host='api.hamsterkombatgame.io', port=443): Max retries exceeded with url: /ip (Caused by ProxyError('Unable to connect to proxy. Your proxy appears to only use HTTP and not HTTPS, try changing your proxy URL to be HTTP. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#https-proxy-error-http-proxy', SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1000)'))))

if i follow the proxy example

**Comment by Fy0urM** [comment]
I honestly don't know, I run the bot on my vps and I have no need to use a proxy. you can read here https://github.com/masterking32/MasterHamsterKombatBot/discussions/133
